### PR TITLE
Use icons from store-icons as blocks (SearchBar)

### PR DESCRIPTION
**Pull request by victorhmp**
#### What problem is this solving?

Allow icons from `vtex.store-icons` to be passed as blocks to the `search-bar` block. That way users can customize each individual icon.

#### How should this be manually tested?

See how weird the icons from the SearchBar look:
[Workspace](https://storeicons--storecomponents.myvtex.com/)

#### Checklist/Reminders

- [x] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes

This depends on: https://github.com/vtex-apps/store-icons/pull/37

### CODE REVIEW

File: docs/SearchBar.md [hunk 1]
```diff
@@ -20,16 +20,17 @@ Then, add `search-bar` block into your app theme, as we do in our [Store Header]
...
+| `autoFocus`           | `Boolean`                               | Identify if the search input should autofocus or not                                                                                              | -             |
+| `maxWidth`            | `Number` \| `String`                    | Max width of the search bar                                                                                                                       | -             |
+| `customSearchPageUrl` | `string`                                | Template for a custom url. It can have a substring `${term}` used as placeholder to interpolate the searched term. (e.g. `/search?query=${term}`) | -             |
+| ~`iconClasses`~       | `String`                                | **DEPRECATED** ~Custom classes for the search icon~ Use the CSS handle `searchBarIcon`.                                                           | -             |
+| ~`iconsProps`~        | `{ "viewBox": String, "size": Number }` | Props to be passed down to the icons from `store-icons` being used by SearchBar.                                                                  | -             |
```
Comment (klzns) [inline docs/SearchBar.md]:
```suggestion
| `iconsProps`        | `{ "viewBox": String, "size": Number }` | Props to be passed down to the icons from `store-icons` being used by SearchBar.                                                                  | -             |
```

File: react/components/SearchBar/components/AutocompleteInput.js [hunk 2]
```diff
@@ -107,6 +108,12 @@ AutocompleteInput.propTypes = {
   iconBlockClass: PropTypes.string,
   /** Identify if the search input should autofocus or not */
   autoFocus: PropTypes.bool,
+  iconsProps: PropTypes.shape({
```
Comment (klzns) [inline react/components/SearchBar/components/AutocompleteInput.js]: 🤔 should we create two separate props for each icon?

Comment (victorhmp) [reply react/components/SearchBar/components/AutocompleteInput.js]: I thought about it... but I think it's better to have a single prop for consistency.
If you set a viewBox for one of the icons, for example, you're setting it for the others as well.

Comment (lbebber) [reply react/components/SearchBar/components/AutocompleteInput.js]: > If you set a viewBox for one of the icons, for example, you're setting it for the others as well
not sure if I get this.

is this behaviour unavoidable? If so, it of course makes sense for both icons to share the same prop, but I don't think this is the ideal behaviour.

File: docs/SearchBar.md [hunk 3]
```diff
@@ -20,16 +20,16 @@ Then, add `search-bar` block into your app theme, as we do in our [Store Header]
...
-| `autoFocus`           | `Boolean`            | Identify if the search input should autofocus or not                                                                                              | -             |
-| `maxWidth`            | `Number` \| `String` | Max width of the search bar                                                                                                                       | -             |
-| `customSearchPageUrl` | `string`             | Template for a custom url. It can have a substring `${term}` used as placeholder to interpolate the searched term. (e.g. `/search?query=${term}`) | -             |
-| ~`iconClasses`~       | `String`             | **DEPRECATED** ~Custom classes for the search icon~ Use the CSS handle `searchBarIcon`.                                                           | -             |
+| Prop name             | Type                                    | Description                                                                                                                                       | Default value |
```
Comment (Klynger) [inline docs/SearchBar.md]: This is so hard to maintain and after some time you can't even see the table formatted in the vscode

Comment (victorhmp) [reply docs/SearchBar.md]: I know! We need a better way to show the list of available props

**Pull request merged**